File: chores/src/main/resources/static/js/app/UserService.js
'use strict'

angular.module('user.service', [])
    .service('UserService', [ "$http", "CONSTANTS", UserService]);


function UserService($http, CONSTANTS) {
    var service = {};
    service.getUserById = function(userId) {
        var url = CONSTANTS.userResource + userId;
        return $http.get(url);
    }
    service.getAllUsers = function() {
        return $http.get(CONSTANTS.userResource);
    }
    service.saveUser = function(userDto) {
        return $http.post(CONSTANTS.userResource, userDto);
    }
    service.deleteUser = function(deleteId) {
        var url = CONSTANTS.userResource + deleteId
        return $http.delete(url);
    }
    return service;
}

File: chores/src/main/java/com/jegner/service/UserService.java
package com.jegner.service;

import java.util.List;

import com.jegner.entity.User;

public interface UserService {
    User getUserById(Integer userId);

    void saveUser(User user);

    List<User> getAllUsers();

    void deleteUser(Integer userId);

    void updateUser(User user);
}
